Kernel/Net/NetworkTask.cpp

```diff
@@ -329,6 +330,38 @@ void flush_delayed_tcp_acks()
     }
 }
 
+void send_tcp_rst(const IPv4Packet& ipv4_packet, const TCPPacket& tcp_packet, RefPtr<NetworkAdapter> adapter)
+{
+    auto routing_decision = route_to(ipv4_packet.source(), ipv4_packet.destination(), adapter);
+    if (routing_decision.is_zero())
+        return;
+
+    auto ipv4_payload_offset = routing_decision.adapter->ipv4_payload_offset();
+
+    const size_t options_size = 0;
+    const size_t tcp_header_size = sizeof(TCPPacket) + options_size;
+    const size_t buffer_size = ipv4_payload_offset + tcp_header_size;
+
+    auto packet = routing_decision.adapter->acquire_packet_buffer(buffer_size);
+    if (!packet)
+        return;
+    routing_decision.adapter->fill_in_ipv4_header(*packet, ipv4_packet.destination(),
+        routing_decision.next_hop, ipv4_packet.source(), IPv4Protocol::TCP,
+        buffer_size - ipv4_payload_offset, 64);
+    memset(packet->buffer.data() + ipv4_payload_offset, 0, sizeof(TCPPacket));
```

Comment:
Using this libc function here is technically not right, I don't think.
Maybe try `__builtin_memset`? that's what the compiler is substituting in anyway